### PR TITLE
feat: review load Icrc accounts and extract common Icrc accounts page

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

This PR addresses two observations:

1.  When the query+update call is activated:

When a user access the accounts, the balance of the universes are fetched with uncertified calls for performance reason. 

However, the selected universe is fetched with query+update for security reason.

Moreover, if a universe is selected, we try to avoid to perform both - i.e. not query+query+update but only query+update.

This PR implements those features for the new Icrc/ETH accounts.

2. Similar to WalletPage we can extract a single common page for all Icrc token accounts related page, that way the logic is reused and ultimately we can refactor Snses

# Notes

The loading of the accounts was adapted but, components wise, no functional changes. The logic and display of the component is extracted to a new component.

# Changes

- use `uncertifiedLoadAccountsBalance` to load balance in token selector and perform same logic to load the information only when needed
- extract `IcrcAccountsPage` from `CkBTCAccounts`
- use this new component in `CkBTCAccounts`
- create `IcrcTokenAccounts` component also use and inherit similar code and logic

### CODE REVIEW

File: frontend/src/lib/routes/Accounts.svelte [hunk 1]
```diff
@@ -69,32 +68,44 @@
...
+  const loadAccountsBalances = async (universeIds: UniverseCanisterIdText[]) => {
+    // Selected universes are empty, no information shall and can be fetched
+    if (isArrayEmpty(universeIds)) {
+      return;
+    }
 
-    return loadIcrcAccounts({ ledgerCanisterIds, certified: false });
+    await uncertifiedLoadAccountsBalance({
+      universeIds,
+      excludeUniverseIds: [selectedUniverseId.toText()],
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/lib/routes/Accounts.svelte:107]: Noted. This PR is still a draft.

[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/routes/Accounts.svelte:107]: 👍

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts [hunk 2]
```diff
@@ -1,61 +1,127 @@
...
+import { render, waitFor } from "@testing-library/svelte";
 
-describe("IcrcTokenAccounts", () => {
-  const ledgerCanisterId = principal(0);
-
-  const renderComponent = () => {
-    const { container } = render(IcrcTokenAccounts);
-
-    return IcrcTokenAccountsPo.under(new JestPageObjectElement(container));
+vi.mock("$lib/services/wallet-accounts.services", () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:24]: There were examples where we mocked the api layer directly. Couldn't you keep that?

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts [hunk 3]
```diff
@@ -1,61 +1,127 @@
...
+        // Do nothing
+      },
+    })
+  ),
+}));
+
+describe("IcrcTokenAccounts", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.spyOn(tokensStore, "subscribe").mockImplementation(
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:46]: Why not just load it into the store?

```
tokensStore.setTokens({
      [ledgerCanisterId.toText()]: {
        certified: true,
        token: mockCkETHToken,
      },
    });
```

We agreed to mock only the layers interacting with the outside world. We don't want to apply it to the old tests, but you're removing setting the store with a mock...

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts [hunk 4]
```diff
@@ -1,61 +1,127 @@
...
-        certified: true,
-      },
+  describe("when no accounts", () => {
+    beforeEach(() => {
+      icrcAccountsStore.reset();
     });
-    const po = renderComponent();
 
-    expect(await po.getAccountCardPos()).toHaveLength(1);
+    it("should call load accounts", () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:114]: It shuold check that the api layer is called.

File: frontend/src/tests/lib/routes/Accounts.spec.ts [hunk 5]
```diff
@@ -343,119 +340,86 @@ describe("Accounts", () => {
...
 
     render(Accounts);
 
-    await runResolvedPromises();
-
-    // It's called once when the component is mounted
-    expect(icrcLedgerApi.queryIcrcToken).toHaveBeenCalledTimes(1);
-    expect(icrcLedgerApi.queryIcrcBalance).toHaveBeenCalledTimes(1);
+    await waitFor(() =>
+      expect(uncertifiedLoadAccountsBalance).toHaveBeenCalled()
```
Comment (lmuntaner) [inline frontend/src/tests/lib/routes/Accounts.spec.ts:359]: You are removing checks to the api for a check on the service...

Can you add a TODO with a link to this issue: https://dfinity.atlassian.net/browse/GIX-2150

File: frontend/src/tests/lib/routes/Accounts.spec.ts [hunk 6]
```diff
@@ -343,119 +340,86 @@ describe("Accounts", () => {
...
     const titleRow = getByTestId("projects-summary");
 
-    expect(
-      titleRow?.textContent?.includes(mockSummary.metadata.name)
-    ).toBeTruthy();
+    expect(titleRow?.textContent?.includes("ckETH")).toBeTruthy();
   });
 
-  it("should render sns project logo", () => {
+  it("should render icrc project logo", () => {
```
Comment (lmuntaner) [inline frontend/src/tests/lib/routes/Accounts.spec.ts:407]: Is this correct? This should be `"should render related icrc project logo"`, right?

**Review by lmuntaner — COMMENTED** [review]
LGTM!

Just a few comments regarding the tests.

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:24]: This is a copy/paste from `CkBTCAccounts.spec.ts`. Feel free to improve both in another PR.

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts (hunk 3 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:46]: This is a copy/paste from `CkBTCAccounts.spec.ts`. Feel free to improve both in another PR.

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:114]: This is already covered by existing tests. What would be the value to duplicate the test here?

File: frontend/src/tests/lib/routes/Accounts.spec.ts (hunk 5 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/routes/Accounts.spec.ts:360]: Sure.

File: frontend/src/tests/lib/routes/Accounts.spec.ts (hunk 6 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/routes/Accounts.spec.ts]: Sure.

File: frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts (hunk 4 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/pages/IcrcTokenAccounts.spec.ts:114]: I have added the similar todo https://dfinity.atlassian.net/browse/GIX-2150

**Review by lmuntaner — APPROVED** [review]

**Pull request merged**